packages/foundry/contracts/oracles/strategy/StrategyOracle.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity >=0.8.13;
+
+import "@yield-protocol/utils-v2/contracts/access/AccessControl.sol";
+import "@yield-protocol/utils-v2/contracts/cast/CastBytes32Bytes6.sol";
+import "../../interfaces/IOracle.sol";
+
+interface IStrategy {
+    /// @notice Returns LP tokens owned by the strategy after the last operation
+    /// @return LP tokens amount
+    function cached() external view returns (uint256);
+
+    /// @notice Returns total supply of the strategy token
+    /// @return Total Supply of strategy token
+    function totalSupply() external view returns (uint256);
+
+    /// @notice Returns baseId of the strategy
+    /// @return baseId
+    function baseId() external view returns (bytes6);
+}
+
+/// @title Oracle contract to get price of strategy tokens in terms of base & vice versa
+/// @author iamsahu
+/// @dev value of 1 LP token = 1 base
+contract StrategyOracle is IOracle, AccessControl {
+    using CastBytes32Bytes6 for bytes32;
+    event SourceSet(
+        bytes6 indexed baseId,
+        bytes6 indexed quoteId,
+        IStrategy indexed strategy
+    );
+
+    struct Source {
+        bool inverse;
+        IStrategy strategy;
+    }
+
+    mapping(bytes6 => mapping(bytes6 => Source)) public sources;
+
+    function setSource(bytes6 strategyId, IStrategy strategy) external auth {
+        bytes6 quoteId = strategy.baseId();
+        sources[strategyId][quoteId] = Source({
+            strategy: strategy,
+            inverse: false
+        });
+        emit SourceSet(strategyId, quoteId, strategy);
+        if (strategyId != quoteId) {
```

Comment:
It would be pretty weird if `strategyId == quoteId`. I would remove this `if` and just store the inverse.